pkg/kapis/cluster/v1alpha1/handler.go

```diff
@@ -251,6 +254,73 @@ func (h *handler) generateDefaultDeployment(cluster *v1alpha1.Cluster, w io.Writ
 	return h.yamlPrinter.PrintObj(&agent, w)
 }
 
+// updateKubeConfig updates the kubeconfig of the specific cluster, this API is used to update expired kubeconfig.
+func (h *handler) updateKubeConfig(request *restful.Request, response *restful.Response) {
+	var req clusterv1alpha1.UpdateClusterRequest
+	if err := request.ReadEntity(&req); err != nil {
+		api.HandleBadRequest(response, request, err)
+		return
+	}
+
+	clusterName := request.PathParameter("cluster")
+	obj, err := h.clusterLister.Get(clusterName)
+	if err != nil {
+		api.HandleBadRequest(response, request, err)
+		return
+	}
+	cluster := obj.DeepCopy()
+	if _, ok := cluster.Labels[v1alpha1.HostCluster]; ok {
+		api.HandleBadRequest(response, request, fmt.Errorf("update kubeconfig of the host cluster is not allowed"))
+		return
+	}
+	// For member clusters that use proxy mode, we don't need to update the kubeconfig,
+	// if the certs expired, just restart the tower component in the host cluster, it will renew the cert.
+	if cluster.Spec.Connection.Type == v1alpha1.ConnectionTypeProxy {
+		api.HandleBadRequest(response, request, fmt.Errorf(
+			"update kubeconfig of member clusters that use proxy mode is not allowed, if the certs expired, just restart the tower component in the host cluster, it will renew the cert",
```

Comment:
Maybe we should automatically renew proxy mod cluster certs in tower, not by restarting tower, it's risky. 
```suggestion
			"update kubeconfig of member clusters which using proxy mode is not allowed, their certs are managed and will be renewed by tower.",
```

Reply:
Sure